### ISSUE TITLE
Cleanup Video field collection data

### ISSUE THREAD

**Issue by aaronschachter** [DoSomethingArchive/legacy-website#969]
Depends on #949 

Purge invalid `field_video` values with no video_id once #949 is in place. Will fix the empty youtube vids on all of the 11 Facts pages.

**Comment by barryclark** [comment]
Another issue related to video field checking.

**Comment by aaronschachter** [comment]
Now that #949 is resolved, I've gone in and manually saved static content nodes to remove the dummy `field_video` records for static content.

However, there's currently 46 records in `field_data_field_video` which are not legit. (the one record for entity_id=349 is legit, it belongs to node 12 which is a test sponsored campaign).  What happened is this: the is_empty check was put in place for field_partners before field_video.  So the field_partners values were removed from the campaigns, but these field_video records are orphans of those removed field_partners values.

This isn't breaking anything, but it would be great to clean them up. Notes on programatically deleting a field collection: http://rajanmayekar.com/blog/programmatically-creating-deleting-modifying-field-collection-item-node
Could run this in an update hook to test.

**Comment by angaither** [comment]
This is a pretty funky issue.

`//Get all records in the table
$result = db_query("SELECT entity_id FROM field_data_field_video");
foreach($result as $nid) {
   $node = node_load($nid->entity_id);
   print "node : " . $nid->entity_id . " : "; print_r($node->type); print "\n";
   $field_collection_item_value = $node->field_video[LANGUAGE_NONE][0]['value'];
   if ($field_collection_item_value) {
      entity_delete_multiple('field_collection_item', array($field_collection_item_value));
  }
}
`
Running this query returns [these results](https://gist.github.com/angaither/47e10c498d9cc85e08ac), which makes it seem like fact nodes had a video field that has been removed.

Also the `entity_delete_multiple` is not working, maybe because they are orphaned?

**Comment by barryclark** [comment]
It's totally possible that facts pages had a video that has now been removed. There's no need for videos on 11 Facts pages any more.

**Comment by aaronschachter** [comment]
This issue is for cleaning up the data in the `field_video` field, where ID's exist for nodes that don't actually have any video set.  Definitely not high priority, but a cleanup rainy day task.
